keras/backend/tensorflow/numpy.py

```diff
@@ -783,6 +788,125 @@ def prod(x, axis=None, keepdims=False, dtype=None):
     return tfnp.prod(x, axis=axis, keepdims=keepdims, dtype=dtype)
 
 
+def _quantile(x, q, axis=None, method="linear", keepdims=False):
+    # ref: tfp.stats.percentile
+    # float64 is needed here and below, else we get the wrong index if the array
+    # is huge along axis.
+    q = tf.cast(q, "float64")
+
+    # Move `axis` dims of `x` to the rightmost, call it `y`.
+    if axis is None:
+        y = tf.reshape(x, [-1])
+    else:
+        x_ndims = tf.TensorShape(x.shape).rank
+
+        # _make_static_axis_non_negative_list
+        axis = list(map(lambda x: x if x >= 0 else x + x_ndims, axis))
+
+        # _move_dims_to_flat_end
+        other_dims = sorted(set(range(x_ndims)).difference(axis))
+        perm = other_dims + list(axis)
+        x_permed = tf.transpose(a=x, perm=perm)
+        if tf.TensorShape(x.shape).is_fully_defined():
```

Comment:
Likewise this could be `if None not in x.shape` to avoid reliance on TensorShape APIs

Reply:
Done